django/db/backends/base/features.py

```diff
@@ -351,6 +351,15 @@ class BaseDatabaseFeatures:
     # Does the backend support column comments in ADD COLUMN statements?
     supports_comments_inline = False
 
+    # Does the backend support generated columns?
+    supports_generated_columns = False
+    # Does the backend allow parameters in generated column definitions?
+    supports_generated_columns_params = False
+    # Does the backend support stored generated columns?
+    supports_stored_generated_columns = False
+    # Does the backend support virtual generated columns?
+    supports_virtual_generated_columns = False
```

Comment:
I'm not sure why we need two feature flags :thinking: `VIRTUAL` in the column definition is optional on Oracle and doesn't change anything. Oracle doesn't support `STORED` virtual columns, is it not enough to set:
```
    supports_generated_columns = True
    supports_generated_columns_params = False
    supports_stored_generated_columns = False
```

Reply:
:thinking: Ah, right. It seems that this time PostgreSQL is responsible for the extra feature flag :wink:. I would leave only:
- `supports_stored_generated_columns`,
- `supports_virtual_generated_columns`, and
- `supports_generated_columns_params`.